lexibank_lieberherrkhobwa.py

```diff
@@ -1,63 +1,88 @@
+            "metroxylon/subgrouping-kho-bwa-v1.0.0.zip"
         )
+        filename = "metroxylon-subgrouping-kho-bwa-333538b/data/dataset_khobwa.csv"
+
+        self.raw.download_and_unpack(zip_url, filename)
+
 
     def split_forms(self, item, value):
+        """
+        Splits and cleans forms before passing them to the tokenizer.
+        """
+
+        # Inform that the value is being overriden
         if value in self.lexemes:  # pragma: no cover
-            self.log.debug('overriding via lexemes.csv: %r -> %r' % (value, self.lexemes[value]))
+            self.log.debug('overriding via lexemes.csv: %r -> %r' %
+                           (value, self.lexemes[value]))
+
+        # Get the replacement value for `lexemes.csv`
         value = self.lexemes.get(value, value)
-        return [self.clean_form(item, form).strip()
-                for form in split_text(value, separators='~/,;')]
 
-    def cmd_install(self, **kw):
+        # Return a list of cleaned forms, splitting over tilde, slash,
+        # comma, and colon
+        forms = [
+            self.clean_form(item, form).strip()
+            for form in split_text(value, separators='~/,;')
+        ]
+
+        return forms
 
+    def cmd_install(self, **kw):
+        # Read the raw data
         data = self.raw.read_csv("dataset_khobwa.csv")
-        assert set(len(r) for r in data) == {201}
-        concept_by_index = OrderedDict()
-        for i in range(1, len(data[0]), 2):
-            concept_by_index[i] = data[0][i]
 
         with self.cldf as ds:
+            # Add bibliographic sources
             ds.add_sources(*self.raw.read_bib())
 
-            # Read raw concept data and append it
-            for concept in self.conceptlist.concepts.values():
-                for i, gloss in list(concept_by_index.items()):
-                    if gloss == concept.english:
-                        concept_by_index[i] = concept.id
-                        break
-                else:
-                    raise ValueError(concept["ENGLISH"])
+            # Read raw concept data and add to dataset; at the same time,
+            # build a map between the concept index in data and the
+            # concept id in the dataset
+            concept_by_index = {}
+            for cid, concept in enumerate(self.conceptlist.concepts.values()):
+                # We can get the concept id from the raw data
+                concept_by_index[1+(cid*2)] = concept.id
 
+                # Add the concept
                 ds.add_concept(
                     ID=concept.id,
                     Name=concept.english,
                     Concepticon_ID=concept.concepticon_id,
                     Concepticon_Gloss=concept.concepticon_gloss,
                 )
 
-            # Read raw languages and append it
+            # Read raw languages and add to the dataset; at the same time,
+            # build a map
             langs = {}
             for language in self.languages:
```

Comment:
you can call `ds.add_languages()` without doing anything if your languages.csv file contains the same NameSpace. I think this should be the case. Check with `chenhmongmien`, for details.

Reply:
Ok.